### ISSUE TITLE
Problem with Userscripts containing Cyrillic symbols

### ISSUE THREAD

**Issue by Futyn-Maker** [quoid/userscripts#227]
_System Information:_
> macOS or iOS version:
iOS 15.4
> Userscripts version:
1.0.2
> Safari version:
Latest
> Is this issue related to script injection?
Partially
> Did the test script (pasted above) successfully run on your machine?
Yes

I'm from Russia and I use userscripts containing Russian (Cyrillic) texts. However, Cyrillic characters displays on the web-page wrong. For example (without the metadata):

```
alert('Привет, мир!');
```
The text 'Привет, мир' displays wrong, with strange symbols.

Scripts is encoded in UTF8.

Can I do something for solving this?

**Comment by quoid** [comment]
@Futyn-Maker 

can you please provide an example userscript that demonstrates the issue you are seeing, *including metadata*? And please include the website where you see the strange symbols.


Below is what I see when using `alert('Привет, мир!');`
<img width="467" alt="Screen Shot 2022-02-06 at 9 50 47 AM" src="https://user-images.githubusercontent.com/7660254/152686537-50269e4e-83f2-4c4b-8b34-76b42e4c0cf9.png">

**Comment by Futyn-Maker** [comment]
@quoid Hello!

Here's the example of script:

```
// ==UserScript==
// @name         RussianScript
// @namespace    http://futyn-maker.ru/
// @version      0.1
// @include        *
// ==/UserScript==

alert('Привет, мир!');
```

I've tested it on many URLs, such as Google, Facebook, some Russian sites (Yandex, VK, my personal server...): the result is similar.

![Screenshot with the example of the output](https://user-images.githubusercontent.com/54239175/152689898-4d70daae-6e44-425f-8f54-0d4f01a12524.PNG)

**Comment by quoid** [comment]
@Futyn-Maker 

Thank you for the example userscript. I am testing it here, but I am not seeing the same as you 🙁 I am on v1.1 `beta` here, but I did not change anything in the way the code is injected from 1.0.2 to 1.1, so I don't *think* that would be why I do not see the issue.

Maybe more information will help:

1) What iPhone are you using?
2) How are you creating and editing your userscripts? What program, etc... (I am wondering if the weird character might be injected in the editor)
3) What language is your iPhone set to?

Below are screenshots from iPhone 8:

<img src="https://user-images.githubusercontent.com/7660254/152691393-81b4ac23-87ac-4ce1-ab4b-2de9b0e77618.jpeg" width="50%" height="50%">


<img src="https://user-images.githubusercontent.com/7660254/152691396-a9939d53-16e7-4c32-a839-7c7ce86ad3be.jpeg" width="50%" height="50%">



<img src="https://user-images.githubusercontent.com/7660254/152691397-3da591ad-12a2-47fc-9eae-323320924477.jpeg" width="50%" height="50%">

Screenshot from iPad Mini 6:

<img src="https://user-images.githubusercontent.com/7660254/152691589-f60b1884-06ec-424a-95a1-491f9d3c1e8f.jpeg" width="50%" height="50%">

**Comment by Futyn-Maker** [comment]
OK, thanks and sorry for this fake issue. I seem to find the problem.

> 2. How are you creating and editing your userscripts? What program, etc... (I am wondering if the weird character might be injected in the editor)

Yes, I've found that It really is. I write my scripts using Notepad++ on my Windows PC, and for some reason iPhone processes Unicode symbols incorectly if the text is encoded in UTF8 (without BOM). I changed the encoding to UTF8 with BOM, and the Russian text became to display correctly.

So it looks like the problem in the text transfer between a Windows PC and my iPhone. Sorry again.
